frontend/packages/component-library/src/carousel/Carousel.tsx

```diff
@@ -0,0 +1,146 @@
+import classNames from 'classnames';
+import {HTMLAttributes, ReactNode} from 'react';
+
+// Import Swiper React components
+// See Swiper documentation here: https://swiperjs.com/react
+import {Swiper, SwiperSlide} from 'swiper/react';
+import {Navigation, Pagination, A11y} from 'swiper/modules';
+
+// Import Swiper styles
+import 'swiper/css';
+import 'swiper/css/navigation';
+import 'swiper/css/pagination';
+
+import moduleStyles from './carousel.module.scss';
+import FontAwesomeV6Icon from '@/fontAwesomeV6Icon/FontAwesomeV6Icon';
+
+export interface CarouselProps extends HTMLAttributes<HTMLElement> {
+  /** Unique identifier for the carousel instance */
+  carouselId?: string;
+  /** Number of slides per view */
+  slidesPerView?: number;
+  /** Number of slides per group */
+  slidesPerGroup?: number;
+  /** Allow touch move */
+  allowTouchMove?: boolean;
+  /** Auto height */
+  autoHeight?: boolean;
+  /** Show navigation arrows */
+  showNavArrows?: boolean;
+  /** Carousel content */
+  slides: {id: string; slide: ReactNode}[];
+}
+
+const SwiperCarousel: React.FC<CarouselProps> = ({
+  carouselId,
+  showNavArrows = true,
+  slidesPerView = 2,
+  slidesPerGroup = 2,
+  allowTouchMove = false,
+  autoHeight = false,
+  slides,
+  className,
+  ...HTMLAttributes
+}: CarouselProps) => {
+  // Constants for carousel configuration
+  const slideGap = 20;
+  const breakpointTablet = 768;
+  const breakpointDesktop = 1024;
+
+  return (
+    <div
+      className={classNames(moduleStyles.carouselWrapper, className)}
+      {...HTMLAttributes}
+    >
+      <div className={classNames(moduleStyles.carousel, className)}>
+        {/* Swiper carousel */}
+        <Swiper
+          modules={[Navigation, Pagination, A11y]}
+          allowTouchMove={allowTouchMove}
+          autoHeight={autoHeight}
+          spaceBetween={slideGap}
+          navigation={{
+            nextEl: `.${carouselId}-next`,
+            prevEl: `.${carouselId}-prev`,
```

Comment:
Optional: I think `#` instead of `.` would be more semantically correct as this would be a unique (once per page) identifier. This would mean using id instead of class.